docs/app-gen.md

```diff
@@ -0,0 +1,145 @@
+# generate
+## Overview
+generate is an experimental application generation utility for OpenShift v3. It is used to create the necessary configuration to build and deploy code from a source repository. 
+
+## Usage
+```
+Experimental command
+
+Generate a new application in OpenShift from source code.
+
+Usage:
+openshift ex generate [source]
+
+The source parameter may be a directory or a repository URL.
+If not specified, the current directory is used.
+
+Examples:
+  $ openshift ex generate
```

Comment:
Please update the markdown here to follow the Github markdown rules eg. when you want to create quoted text which contains shell scripts, leave its previous and next lines empty (so there should be an empty line between `Examples` and the first command here). Also use 4 spaces instead of 2 for indentation. See this [fix](https://github.com/GoogleCloudPlatform/kubernetes/pull/4065) on Kubernetes for `kubectl.md`.